src/rpc/client_channel.rs

```diff
@@ -205,13 +204,6 @@ impl WebRTCClientChannel {
         // header math in the process.
 
         let mut to_add_bytes = [0u8; 4];
-        // 1-5 because those are the length header bytes for gRPC
-        to_add_bytes.clone_from_slice(&data[1..5]);
-        let mut next_message_length = u32::from_be_bytes(to_add_bytes);
-        // if this is all streaming calls we need to tell the server when we're done with
-        // the stream, otherwise neither side will know we're done, trailers will never be
-        // sent/processed, and we'll hang on the strream.
-        let it_was_all_a_stream = usize::try_from(next_message_length).unwrap() + 5 < data.len();
```

Comment:
All of this logic to determine whether or not we were in a stream was only done once, at the top level loop. This made sense insofar as it would accurately tell us if we were in a streaming scenario (a fact that wouldn't change), but it wasn't ideal for determining when we were at end of stream (which we were using it for).